### PR TITLE
Use Github actions for builds

### PR DESCRIPTION
**Pull request by ilbambino**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,8 +13,8 @@ The [`CARTO Maps API`](https://carto.com/developers/maps-api/) tiler. It extends
 
 Requirements:
 
-* [`Node 10.x (npm 6.x)`](https://nodejs.org/dist/latest-v10.x/)
-* [`PostgreSQL >= 10.0`](https://www.postgresql.org/download/)
+* [`Node 12.x `](https://nodejs.org/dist/latest-v10.x/)
```
Comment (dgaubert) [inline README.md:16]:
```suggestion
* [`Node 12.x `](https://nodejs.org/dist/latest-v12.x/)
```

File: lib/api/api-router.js [hunk 2]
```diff
@@ -199,8 +199,8 @@ module.exports = class ApiRouter {
             const apiRouter = router({ mergeParams: true });
             const { paths, middlewares = [] } = route;
 
-            apiRouter.use(user(this.metadataBackend));
             apiRouter.use(initLogger({ logger: this.serverOptions.logger }));
+            apiRouter.use(user(this.metadataBackend));
```
Comment (dgaubert) [inline lib/api/api-router.js:203]: I'm curious, is this a leftover during the process of getting this PR done?

File: package.json [hunk 3]
```diff
@@ -94,8 +95,7 @@
     "posttest": "npm run test:teardown",
     "test:teardown": "NODE_ENV=test node test teardown",
     "cover": "nyc --reporter=lcov npm test",
-    "test:docker": "docker run -e \"NODEJS_VERSION=$NODE_VERSION\" -v `pwd`:/srv $DOCKER_IMAGE bash docker/scripts/test-setup.sh && docker ps --filter status=dead --filter status=exited -aq | xargs docker rm -v",
-    "docker:bash": "docker run -it -v `pwd`:/srv $DOCKER_IMAGE bash"
+    "dev": "NODE_ENV=development nodemon app.js"
```
Comment (dgaubert) [inline package.json:98]: When this PR is merged I guess [this](https://github.com/CartoDB/docker-dev-env/blob/master/docker-compose.yml#L99)  should be updated, right?

File: test/index.js [hunk 4]
```diff
@@ -25,28 +28,28 @@ const PUBLIC_USER_PASSWORD = environment.postgres.password;
 const TEST_DB = `${TEST_USER}_db`;
 
 async function startRedis () {
-    await exec(`redis-server --port ${REDIS_PORT} --loadmodule ${REDIS_CELL_PATH} --logfile ${__dirname}/redis-server.log --daemonize yes`);
+    // await exec(`redis-server --port ${REDIS_PORT} --loadmodule ${REDIS_CELL_PATH} --logfile ${__dirname}/redis-server.log --daemonize yes`);
```
Comment (dgaubert) [inline test/index.js:31]: we can remove the function completely... If we need it back again we always can revert this change

File: test/index.js [hunk 5]
```diff
@@ -25,28 +28,28 @@ const PUBLIC_USER_PASSWORD = environment.postgres.password;
...
 }
 
 async function stopRedis () {
-    await exec(`redis-cli -p ${REDIS_PORT} shutdown`);
+    // await exec(`redis-cli -p ${REDIS_PORT} shutdown`);
```
Comment (dgaubert) [inline test/index.js:35]: Same as the previous comment.

**Review by dgaubert — APPROVED** [review]
Impressive! 👏 

Left some minor comments, nothing blocking.

File: lib/api/api-router.js (hunk 2 above)
Comment (ilbambino) [reply lib/api/api-router.js:203]: nope, it is a bug that bit me. It was crashing in some weird case (DB not responding). And as the logger was not init, it was crashing (inside the `user` middleware). Changing the order fixed it
